ext/flash/sendfile.rs

```diff
@@ -35,6 +138,42 @@ impl SendFile {
     {
       // Send all bytes.
       let mut length = 0;
+
+      if self.sending_headers {
+        let mut hdtr = libc::sf_hdtr {
+          headers: self.slices.as_mut_ptr() as _,
+          hdr_cnt: self.slices.len() as libc::c_int,
+          trailers: std::ptr::null_mut(),
+          trl_cnt: 0,
+        };
+        // On macOS `length` is value-result parameter. It determines the number
+        // of bytes to write and returns the number of bytes written also in
+        // case of `EAGAIN` errors.
+        // SAFETY: call to libc::sendfile()
+        let res = unsafe {
+          libc::sendfile(
+            self.io.0,
+            self.io.1,
+            self.written as libc::off_t,
+            &mut length,
+            &mut hdtr,
+            0,
+          )
+        };
+        if res == -1 {
```

Comment:
You should special-case EAGAIN and EINTR here:

- EAGAIN and length == 0, treat as regular EAGAIN
- EAGAIN and length > 0, some (possibly all) data was written

Ditto for EINTR.